src/primitives/transaction.cpp

```diff
@@ -161,18 +160,26 @@ std::vector<char> loadTransaction(const std::vector<CMFToken> &tokens, std::vect
             seenCoinbaseMessage = true;
             break;
         }
-        case Consensus::TxInScript: {
-            if (inputs.empty()) throw std::runtime_error("TxInScript before TxInPrevHash");
-            if (inputCount == 0) { // copy all of the input tags
+        case Consensus::TxInputStackItem:
+        case Consensus::TxInputStackItemContinued: {
+            if (signatureCount == -1) { // copy all of the input tags
                 CDataStream stream(0, 4);
                 ser_writedata32(stream, nVersion);
                 for (unsigned int i = 0; i < index; ++i) {
                     tokens[i].Serialize(stream, 0, 4);
                 }
                 txData = std::vector<char>(stream.begin(), stream.end());
             }
+            if (signatureCount < 0 || token.tag == Consensus::TxInputStackItem)
+                signatureCount++;
+            if (static_cast<int>(inputs.size()) <= signatureCount)
+                throw std::runtime_error("TxInputStackItem* before TxInPrevHash");
+
             auto data = token.unsignedByteArray();
-            inputs[inputCount++].scriptSig = CScript(data.begin(), data.end());
+            if (data.size() == 1)
+                inputs[signatureCount].scriptSig << data.at(0);
+            else
+                inputs[signatureCount].scriptSig << data;
```

Comment:
Had the same problem. Figured its possible to avoid << by using insert, see: https://github.com/dagurval/bitcoinxt/commit/2e5cef1eb58c6bd42a6ec1990020bbc414de985a#diff-cd7b305fd4b4280f22ae88960e60398eR64